docs/syscall_descriptions.md

```diff
@@ -127,6 +127,65 @@ change in descriptions for a particular syscall, the programs that are already i
 the corpus will be kept there, unless you manually clear them out (for example by
 removing the `corpus.db` file).
 
+<div id="tips"/>
+
+## Description tips and FAQ
+
+<div id="names"/>
+
+### Syscall, struct, field, flags names
+
+Stick with existing kernel names for things, don't invent new names if possible.
+
+For example, if there is an existing enum `v4l2_buf_type` in the kernel headers,
+use this name for flags in descriptions as well. The same for structs, unions,
+fields, etc. For syscall variants, use the command name after `$` sign.
+For example, `fcntl$F_GET_RW_HINT`, `ioctl$FIOCLEX`, `setsockopt$SO_TIMESTAMP`.
+
+This makes it easier to look up things in the kernel sources; make things look
+familiar for people who already know the kernel interface; makes things consistent;
+enables static checking of descriptions (e.g. missed flags or mis-typed fields)
+with [syz-check](/tools/syz-check/check.go) utility.
+
+<div id="ordering"/>
+
+### Resources for syscall ordering
+
+Resources and resource directions (`in`, `out`, `inout`) impose implicit ordering
+constraints on involved syscalls.
+
+If a syscall accepts a resource of a particular type (e.g. has `fd_cdrom` as an input),
+then it will be generally placed after a syscall that has this resource as output,
+so that the resource value can be passed between syscalls. For example:
+
+```
+r0 = openat$cdrom(...)
+ioctl$CDROMPAUSE(r0, 0x123)
+close(r0)
+```
+
+Syscall arguments are always `in`, return values are `out` and pointer indirections
+have explicit direction as `ptr` type attribute.
+
+<div id="values"/>
+
+### Use of unexpected/undeclared values
+
+When specifying integer/string flags or integer fields stick with the official expected values only.
+
+It's tempting to add some unexpected undeclared values to descriptions (e.g. `-1` or `INT_MAX`).
+In the end bugs are usually triggered by unexpected values. This is not encouraged for several reasons.
```

Comment:
This doesn't flow.

I would propose shuffling this a bit (simple fact -> making the leap to the resulting pitfall -> don't do it):
```
Commonly, bugs are triggered by unexpected inputs. With that in mind, it can be too tempting to introduce some unexpected values to descriptions (e.g. `-1` or `INT_MAX`). This is not encouraged for several reasons.
.......
```

Reply:
This and the other suggestions done. Thanks.